node/test/mocktests.ts

```diff
@@ -183,5 +184,100 @@ describe('Mock Tests', function () {
+        const expectedStdout = "atool output here" + os.EOL + "abc";
+        const expectedStderr = "atool with this stderr output" + os.EOL + "def";
+        var a: ma.TaskLibAnswers = <ma.TaskLibAnswers>{
+            "exec": {
+                "/usr/local/bin/atool --arg foo": {
+                    "code": 0,
+                    "stdout": expectedStdout,
+                    "stderr": expectedStderr
+                }
+            }
+        };
+
+        mt.setAnswers(a);
+
+        let tool: mtr.ToolRunner = mt.tool('/usr/local/bin/atool');
+        tool.arg('--arg');
+        tool.arg('foo');
+
+        let firstStdline = true;
+        let firstErrline = true;
+        let numStdLineCalls = 0;
+        let numStdErrCalls = 0;
+        tool.on('stdout', (out) => {
+            assert.equal(expectedStdout, out);
+        });
+        tool.on('stderr', (out) => {
+            assert.equal(expectedStderr, out);
+        });
+        tool.on('stdline', (out) => {
+            numStdLineCalls += 1;
+            if (firstStdline) {
+                assert.equal("atool output here", out);
+                firstStdline = false;
+            }
+            else {
+                assert.equal("abc", out);
+            }
+        });
+        tool.on('errline', (out) => {
+            numStdErrCalls += 1;
+            if (firstErrline) {
+                assert.equal("atool with this stderr output", out);
+                firstErrline = false;
+            }
+            else {
+                assert.equal("def", out);
+            }
+        });
+        await tool.exec(<mtr.IExecOptions>{});
+        
+        assert.equal(numStdLineCalls, 2);
+        assert.equal(numStdErrCalls, 2);
+    })
+    
+    it('Mock toolRunner returns correct output when ending on EOL', async () => {
+        const expectedStdout = os.EOL;
+        const expectedStderr = os.EOL;
+        var a: ma.TaskLibAnswers = <ma.TaskLibAnswers>{
+            "exec": {
+                "/usr/local/bin/atool --arg foo": {
+                    "code": 0,
+                    "stdout": expectedStdout,
+                    "stderr": expectedStderr
+                }
+            }
+        };
+
+        mt.setAnswers(a);
+
+        let tool: mtr.ToolRunner = mt.tool('/usr/local/bin/atool');
+        tool.arg('--arg');
+        tool.arg('foo');
+        let numStdLineCalls = 0;
+        let numStdErrCalls = 0;
+        tool.on('stdout', (out) => {
+            assert.equal(expectedStdout, out);
+        });
+        tool.on('stderr', (out) => {
+            assert.equal(expectedStderr, out);
+        });
+        tool.on('stdline', (out) => {
+            numStdLineCalls += 1;
+            assert.equal("", out);
+        });
+        tool.on('errline', (out) => {
+            numStdErrCalls += 1;
+            assert.equal("", out);
+        });
+        await tool.exec(<mtr.IExecOptions>{});
+        
+        assert.equal(numStdLineCalls, 2);
```

Comment:
Based on my reading of `processLineBuffer`, I think these would be 1 for the real tool runner.